src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -220,20 +220,23 @@ one_shot_object_detector::one_shot_object_detector() {
   model_.reset(new turi::object_detection::object_detector());
 }
 
-gl_sframe one_shot_object_detector::augment(const gl_sframe &data,
-                                            const std::string& image_column_name,
-                                            const std::string& target_column_name,
-                                            const gl_sarray &backgrounds,
-                                            std::map<std::string, flexible_type> &options){
-
-  // TODO: Automatically infer the image column name, or throw error if you can't
-  // This should just happen on the Python side.
-  gl_sframe augmented_data = data_augmentation::augment_data(data,
-                                                             image_column_name,
-                                                             target_column_name,
-                                                             backgrounds,
-                                                             options["seed"],
-                                                             options["verbose"]);
+gl_sframe one_shot_object_detector::augment(
+    const gl_sframe& data, const std::string& image_column_name,
+    const std::string& target_column_name, gl_sarray& backgrounds,
+    std::map<std::string, flexible_type>& options) {
+  gl_sarray small_backgrounds = backgrounds.apply(
+      [](const flexible_type& bg) {
+        flex_image background = bg.get<flex_image>();
+        return image_util::resize_image(bg, background.m_width / 2,
+                                        background.m_height / 2,
```

Comment:
This code path also applies to user-provided backgrounds too, right? How do we know that halving these is the correct thing to do?

Reply:
Oh good point, will fix.

Reply:
Fixed!